### DISCUSSION TITLE
RDFLib 6.2.0 release planning

### DISCUSSION THREAD

**Discussion by aucampia** [RDFLib/rdflib#1803] [Development]
Starting this discussion so we can stipulate what should be in 6.2.0 - please add any things you would like to see before the release as comments and then we can try make issues for it and try and implement it all.

CC: @gjhiggins @nicholascar @white-gecko @ashleysommer

**Comment by aucampia** [comment]
I think we should aim for a release more or less on 2022-06-01 - I would like earlier but I think that is most realistic. Ideally we should tag all issues we think we can fix within this time-frame.

**Reply by ghost** [reply]
> I think we should aim for a release more or less on 2022-06-01 - I would like earlier but I think that is most realistic. Ideally we should tag all issues we think we can fix within this time-frame.

How about [2022-06-04](https://github.com/RDFLib/rdflib/blob/master/CHANGELOG.md#2002-06-04-release-090)? It would coincide with the 20th. anniversary of the initial release of RDFLib 0.9 after being split from redfootlib.

Also, because of the changes in API resulting from the "identifier-as-context-and-remove-ConjunctiveGraph" work, perhaps we should consider a 7.0 release?

**Reply by aucampia** [reply]
The date I had in mind was more considering just fixing a couple of bugs, and maybe getting https://github.com/RDFLib/rdflib/pull/1686 in, and maybe some typing, basic quality of life stuff which I think is having a detrimental effect on our users. I think if we want to include https://github.com/RDFLib/rdflib/pull/1814 this will not be realistic, though I may be wrong. I also think that it may be good to at least have everything we plan to deprecate marked as deprecated in at least one release before removing them, but I can also live without this.

**Reply by ghost** [reply]
No worries, just an off-the-wall suggestion.

**Reply by aucampia** [reply]
I'm up for it though, depending on how others feel, though I think there are many things we want to get into 7.0 - the other approach is to basically just kind of ramp up the interface breaking instead of waiting, though we may end up releasing a new major version every quarter if we do this.

**Comment by nicholascar** [comment]
I think we had previously sort of decided that changes to the graph identifiers would result in a 7.0.0 release, not a 6.x. 

Other things to include (and I think these will easily be included, perhaps before 7.0.0 in a 6.1.x or a 6.x release due to current PRs) are:

* completion of prefix binding issues
* completion of test reorg & modernisation
* _add more in here @nicholascar_

We also have work on RDF* going on now ([rdflib-rdfstar](https://github.com/RDFLib/rdflib-rdfstar)) and something _may_ be ready for 7.0.0 release...

**Reply by ghost** [reply]
> I think we had previously sort of decided that changes to the graph identifiers would result in a 7.0.0 release, not a 6.x.

That was my understanding, _if_ it makes the cut.

**Reply by aucampia** [reply]
> I think we had previously sort of decided that changes to the graph identifiers would result in a 7.0.0 release, not a 6.x.

I agree on this point, was thinking before this, I would like to get into a monthly, or at least quarterly release cadence if possible. We can maybe hold off on releases somewhat but we should also work on reducing the effort to make a release.

**Reply by nicholascar** [reply]
I definitely support a quarterly release cadence! I think monthly's a bit much but certainly having more frequent releases will reduce the effort needed for each release. The effort has reduced a lot with 5.0.0 > 6.0.0 > 6.0.1 > 6.1.0 > 6.1.1 effort, but we still do have a _lot_ of stuff in master not yet in a release.

**Reply by nicholascar** [reply]
@gjhiggins can you please email me on nicholas.car@surroundaustralia.com so I can send you emails?

**Reply by ghost** [reply]
I'm `gjh@bel-epa.com`, your email addy is whitelisted.
